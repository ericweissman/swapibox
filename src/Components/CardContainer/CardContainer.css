/* No CSS *//*# sourceMappingURL=CardContainer.css.map */