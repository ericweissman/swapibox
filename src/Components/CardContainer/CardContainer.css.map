{
    "version": 3,
    "mappings": "",
    "sources": [
        "CardContainer.scss"
    ],
    "names": [],
    "file": "CardContainer.css"
}