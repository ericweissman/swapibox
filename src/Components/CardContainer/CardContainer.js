import React from 'react';
import Card from '../Card/Card'
import '../../Main.scss';
import PropTypes from 'prop-types'
import { uid } from 'uid'

const CardContainer = (props) => {
  let { active, category } = props;
  if (active !== '' && category[active].length > 0) {
    return(
      <div className='card-container'>
        <Card 
          key={uid}
          active={active}
          category={category}
        />
      </div>
    )
  } else if (active !== '' && category[active].length === 0) {
    return(
      <div className='card-container loading'>
        <h2>Loading</h2>
      </div>
    )
  } else {
      return (
        <div className='card-container choose'>
          <h2>chose a category, you must...</h2>
        </div>
      )}
  }

CardContainer.propTypes = {
  active: PropTypes.string,
  category: PropTypes.shape({
    films: PropTypes.object,
    people: PropTypes.array,
    planets: PropTypes.array,
    vehicles: PropTypes.array,
    active: PropTypes.string,
    errorStatus: PropTypes.errorStatus
  })
}

export default CardContainer;