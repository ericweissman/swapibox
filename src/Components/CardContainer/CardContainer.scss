.card-container {
  font-family: $text-font;
    color: $accent;
    display: flex;
    flex-wrap: wrap;
    justify-content: center;
    padding: 20px 10px;
}

.choose {
  h2 {
    font-size: 35px;
    text-transform: uppercase;
  }
}

.loading {
  animation: ySpinLoad 2.6s linear 0s infinite normal; 
  animation-fill-mode: forwards;
  @extend .choose
}

@keyframes ySpinLoad {
  0% {
    transform: rotateY(0deg);
  }
  100% {
    transform: rotateY(360deg);
  }
}