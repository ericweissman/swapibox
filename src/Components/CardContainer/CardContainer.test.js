import React from 'react';
import CardContainer from './CardContainer';
import { shallow } from 'enzyme';

describe('CardContainer', () => {
  let wrapper;
  let mockActive;
  let mockEmpty = {people: [], planets: [], vehicles: []}
  let mockFull = {
    people: [{ name: 'a', homeworld: 'b', population: 'c', species: 'd' }],
    planets: [{ name: 'a', terrain: 'b', population: 'c', climate: 'd', residents: ['lile'] }],
    vehicles: [{ name: 'a', class: 'b', passengers: 'c' }]
  }

  it('should match the correct snapshot for onload', () => {
    mockActive = ''
    wrapper = shallow(<CardContainer active={mockActive} category={mockEmpty}/>)
    expect(wrapper).toMatchSnapshot();
  });

  it('should match the correct snapshot when a category is selected but it is still fetching', () => {
    mockActive = 'people'
    wrapper = shallow(<CardContainer active={mockActive} category={mockEmpty} />)
    expect(wrapper).toMatchSnapshot();
  });

  it('should match the correct snapshot when a category is selected and fetching is complete', () => {
    mockActive = 'people'
    wrapper = shallow(<CardContainer active={mockActive} category={mockFull} />)
    expect(wrapper).toMatchSnapshot();
  });
   
})