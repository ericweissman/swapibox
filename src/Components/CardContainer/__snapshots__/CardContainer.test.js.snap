// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer should match the correct snapshot for onload 1`] = `
<div
  className="card-container choose"
>
  <h1>
    CHOOSE A CATEGORY
  </h1>
</div>
`;

exports[`CardContainer should match the correct snapshot when a category is selected and fetching is complete 1`] = `
<div
  className="card-container"
>
  <Card
    active="people"
    category={
      Object {
        "people": Array [
          Object {
            "homeworld": "b",
            "name": "a",
            "population": "c",
            "species": "d",
          },
        ],
        "planets": Array [
          Object {
            "climate": "d",
            "name": "a",
            "population": "c",
            "residents": Array [
              "lile",
            ],
            "terrain": "b",
          },
        ],
        "vehicles": Array [
          Object {
            "class": "b",
            "name": "a",
            "passengers": "c",
          },
        ],
      }
    }
  />
</div>
`;

exports[`CardContainer should match the correct snapshot when a category is selected but it is still fetching 1`] = `
<div
  className="card-container loading"
>
  <h1>
    Loading
  </h1>
</div>
`;
